viewer/src/views/threejs/cad/RootSectorNode.ts

```diff
@@ -2,58 +2,77 @@
  * Copyright 2020 Cognite AS
  */
 
-import { Sector, SectorQuads, SectorMetadata } from '../../../models/cad/types';
+import * as THREE from 'three';
 import { buildScene } from './buildScene';
 import { SectorNode } from './SectorNode';
 import { toThreeMatrix4 } from '../utilities';
 import { CadModel } from '../../../models/cad/CadModel';
-import { consumeSectorDetailed } from './consumeSectorDetailed';
-import { consumeSectorSimple } from './consumeSectorSimple';
-import { findSectorMetadata } from '../../../models/cad/findSectorMetadata';
-import { discardSector } from './discardSector';
 import { Shading } from './shading';
+import { MemoryRequestCache } from '../../../cache/MemoryRequestCache';
+import { ParsedSector } from '../../../data/model/ParsedSector';
+import { ConsumedSector } from '../../../data/model/ConsumedSector';
+import { LevelOfDetail } from '../../../data/model/LevelOfDetail';
+import { flatMap } from 'rxjs/operators';
+import { consumeSectorSimple } from './consumeSectorSimple';
+import { consumeSectorDetailed } from './consumeSectorDetailed';
+import { SectorQuads, Sector } from '../../../models/cad/types';
+
+function hashIdAndLevelOfDetail(id: number, levelOfDetail: LevelOfDetail) {
+  return `${id},${levelOfDetail}`;
+}
 
 export class RootSectorNode extends SectorNode {
-  private readonly sectorNodeMap: Map<number, SectorNode>;
-  private readonly shading: Shading;
-  private readonly rootSectorMetadata: SectorMetadata;
+  public readonly sectorNodeMap: Map<number, SectorNode>;
+  public readonly shading: Shading;
+
+  private readonly consumeSectorCache: MemoryRequestCache<string, ParsedSector, THREE.Group>;
```

Comment:
I didn't know this had it's own cache :O

Reply:
Discussed already :)